src/generate.c

```diff
@@ -300,10 +302,11 @@ int main(int argc, char** argv)
     if (netplan_state_get_backend(np_state) == NETPLAN_BACKEND_NM || any_nm)
         _netplan_g_string_free_to_file(g_string_new(NULL), rootdir, "/run/NetworkManager/conf.d/10-globally-managed-devices.conf", NULL);
 
+    gboolean enable_wait_online = _netplan_networkd_write_wait_online(np_state, rootdir);
```

Comment:
Note: we should consider writing the s-n-wait-online override only if we have "any_networkd", to avoid fiddling with networkd when nothing is configured through Netplan.